### ISSUE TITLE
Add docker healthcheck

### ISSUE THREAD

**Issue by Bassadin** [jonhadfield/soba#26]
Not sure how relevant it is for this project, but I'd love to have a healthcheck in the dockerfile :)

https://docs.docker.com/engine/reference/builder/#healthcheck

Example: https://github.com/Soluto/golang-docker-healthcheck-example

**Comment by jonhadfield** [comment]
Hi, What use-case are you thinking of?  
Not sure I can see the benefit whilst it's not listening on a port, like a webserver.

**Comment by Bassadin** [comment]
I'm not sure if it can be done, but I'd love to know if the service/"cronjob" is still running. Not sure if it always exists with an error code if there is an issue?
If it's not really necessary, feel free to close ^^

**Comment by jonhadfield** [comment]
Makes sense. I run the docker image on a Synology NAS that you can configure to Auto Restart if the process ends.
One approach to running with cron is to use flock: https://serverfault.com/a/82863/463326.  You could set the cron to run every minute (as in the serverfault example) so it automatically restarts if it fails and can check the existence of the lock file to know if it's running.

soba will exit with `1` if there's an error and `0` if you run it once (without a backup interval) and it succeeds.

**Comment by Bassadin** [comment]
Oh, sorry, I think I phrased that the wrong way.
I'm also using the docker image on a Synology NAS (together with Portainer, but that's not important I presume) and I just set the GIT_BACKUP_INTERVAL to a certain hours value (that's what I meant with "cronjob").
However, I think you're right, just being on the lookout for that `1` error code is probably enough. Seems like I didn't think this through.
Thanks a lot for your extensive answer anyway! I love this image, keep up the awesome work <3
